appinventor/appengine/src/com/google/appinventor/client/editor/simple/SimpleVisibleComponentsPanel.java

```diff
@@ -20,11 +21,14 @@
 import com.google.gwt.user.client.ui.VerticalPanel;
 import com.google.gwt.user.client.ui.Widget;
 
+import java.util.List;
+import java.util.Map;
+
 /**
  * Panel in the Simple design editor holding visible Simple components.
  *
  */
-public final class SimpleVisibleComponentsPanel extends Composite implements DropTarget {
+public final class SimpleVisibleComponentsPanel extends Composite implements DropTarget, ComponentDatabaseChangeListener {
   // UI elements
```

Comment:
The ComponentDatabaseChangeListener implementation does nothing, but it serves as an example for Panels that could change upon a Component Upgrade.

The CDK for Visible components could probably use this for changing the MockComponent.